src/decode.rs

```diff
@@ -1923,6 +1924,8 @@ unsafe fn decode_b_inner(
                 TileStateRef::Frame => &f.lf.lvl,
                 TileStateRef::Local => &ts.lflvlmem,
             };
+            let mut a_uv_guard;
+            let mut l_uv_guard;
```

Comment:
I'm not sure you need these guards, won't passing the mutable references as parameters keep the guards alive for the same lifetime?

Reply:
These guards are for the second pair of slices that get passed in as a tuple. Because of the way that the tuple is being created inline in the function call we need to explicitly create the guards.